pkg/event-driver/drivers/vcenter/vcenter.go

```diff
@@ -148,19 +148,25 @@ func (d *vCenterDriver) processEvent(e types.BaseEvent) (*events.Event, error) {
 	return d.dispatchEvent(topic, ve)
 }
 
-func (d *vCenterDriver) dispatchEvent(topic string, ve *vCenterEvent) (*events.Event, error) {
+func (d *vCenterDriver) dispatchEvent(topic string, ve *vCenterEvent) (*events.CloudEvent, error) {
 	defer trace.Tracef("topic: %s", topic)()
 
 	encoded, err := json.Marshal(*ve)
 	if err != nil {
 		return nil, err
 	}
 
-	event := events.Event{
-		Topic:       topic,
-		ID:          uuid.NewV4().String(),
-		Body:        encoded,
-		ContentType: "application/json",
+	event := events.CloudEvent{
+		Namespace:          "vcenter.vmware.com",
+		EventType:          topic,
+		EventTypeVersion:   "0.1",
+		CloudEventsVersion: events.CloudEventsVersion,
+		SourceType:         "vcenter",
+		SourceID:           "vcenter1",
```

Comment:
is source ID unique per event type or driver instance?

Reply:
The SourceID here is for testing, I haven't had time yet to retrieve it from vCenter but it should be different, e.g. based on the hostname.
But this is one of the things I would like to discuss. CloudEvent spec specifies SourceID and SourceType as mandatory. They describe the source of the event.

Now, the driver is created with a certain name. It's possible that two drivers (with different names) will be created for the same source (e.g. same vcenter). 

The driver is expected to feed CloudEvent-formatted to the sidecar. it picks the SourceID and sends to Sidecar. Now, sidecar must decide how to route that event. My idea was that event is routed based on 3 things: SourceType (e.g. vcenter), SourceID (e.g. vcenter1.corp.local), and EventType (e.g. `vm.created`). To ensure uniqueness in routing, I was thinking about overwriting SourceID with the name of the driver - this name is exposed to the sidecar at the time it's provisioned, and would ensure uniqueness (as name are unique per org, and different org would use different exchange in RabbitMQ). However, this would cause original SourceID coming from vcenter to be lost.
I think the best would be if I could make the routing decision based on the driver name, but somehow convey this without overwriting Source ID, so that when a function receives the event, Source ID remains the same. I will look into that.